### ISSUE TITLE
[release/3.1][release/5.0] Source-build does not build on macOS 10.15 or macOS 11

### ISSUE THREAD

**Issue by lbussell** [dotnet/source-build#2703]
See [this CI run](https://github.com/dotnet/source-build/pull/2701/checks?check_run_id=4589217942
) for an example of where this happens. The build fails while building corefx native components for macOS with the error:
```
EXEC(0,0): error : include location '/usr/local/include' is unsafe for cross-compilation [-Werror,-Wpoison-system-directories]
```

Source-build also fails on macOS 11 for a different error, but the CI run where I observed that result was not retained.

**Comment by lbussell** [comment]
I have fixed the macOS 10.15 issues on both .NET 3.1 and 5.0 with https://github.com/dotnet/source-build/pull/2701 and https://github.com/dotnet/source-build/pull/2702, so our CI is unblocked.

I'm closing this now, if we ever run into specific issues with macOS 11/12 then we can open another issue or re-open this one.
